app/src/main/java/app/grapheneos/camera/App.kt

```diff
@@ -0,0 +1,132 @@
+package app.grapheneos.camera
+
+import android.Manifest
+import android.app.Application
+import android.content.pm.PackageManager
+import android.location.Location
+import android.location.LocationListener
+import android.location.LocationManager
+import androidx.annotation.RequiresPermission
+import app.grapheneos.camera.ui.activities.MainActivity
+
+class App : Application() {
+
+    private var activity: MainActivity? = null
+    private var location: Location? = null
```

Comment:
same here. something like LocationDelegate or LocationManager which will give us an access to get data